content/lessons/chapter-3/solo-1.tsx

```diff
@@ -59,8 +77,43 @@ export default function Solo1({ lang }) {
     setAntagonistsBlockAmount(newBlock)
   }
 
+  useEffect(() => {
+    let interval: NodeJS.Timeout
+    let currentProtagonistHash = protagonistHash
+    if (step === 1) {
+      interval = setInterval(() => {
+        currentProtagonistHash =
+          currentProtagonistHash +
+          Math.floor(Math.random() * PROTAGONISTS[0].hashpower)
+        setProtagonistHash(currentProtagonistHash)
+      }, 40)
+    }
+    return () => clearInterval(interval)
+  }, [step])
+
+  useEffect(() => {
+    let interval: NodeJS.Timeout
+    let currentAntagonistHash = antagonistHash
+    if (step === 1) {
+      interval = setInterval(() => {
+        currentAntagonistHash =
+          currentAntagonistHash +
+          Math.floor(Math.random() * ANTAGONISTS[0].hashpower)
+        setAntagonistHash(currentAntagonistHash)
+      }, 40)
+    }
+    return () => clearInterval(interval)
+  }, [step])
```

Comment:
You can combine this into a single `useEffect` since they have the same dependencies.